src/Microsoft.Android.Build.BaseTasks/AsyncTaskExtensions.cs

```diff
@@ -79,28 +107,47 @@ public static ParallelLoopResult ParallelForEachWithLock<TSource> (this AsyncTas
 			});
 		}
 
-		static ParallelOptions ParallelOptions (AsyncTask asyncTask) => new ParallelOptions {
+		static ParallelOptions ParallelOptions (AsyncTask asyncTask, int maxConcurrencyLevel) => new ParallelOptions {
 			CancellationToken = asyncTask.CancellationToken,
-			TaskScheduler = TaskScheduler.Default,
+			TaskScheduler = GetTaskScheduler (maxConcurrencyLevel),
 		};
 
+		static TaskScheduler GetTaskScheduler (int maxConcurrencyLevel)
+		{
+			var pair = new ConcurrentExclusiveSchedulerPair (TaskScheduler.Default, maxConcurrencyLevel);
+			return pair.ConcurrentScheduler;
+		}
+
+		static int DefaultMaxConcurrencyLevel => Environment.ProcessorCount * 2;
```

Comment:
I think you actually want to reduce this given the _LongRunning_. 

LongRunning == fire up a new worker thread just for the task, assuming AOT is CPU heavy, given a 4 core machine this will fire up 8 threads all competing for CPU. I think you'd be better off reducing this down to the same or just less than the ProcessorCount.